[Terra-form-single-select] Fix Keyboard interaction for form-single-select (#3906)
cerner/terra-core: packages/terra-form-select/CHANGELOG.md

# Changelog

## Unreleased
* Fixed

>   * Fixed selection did not work with space-key `form-single-select`.

Reviewer: ```suggestion
  * Fixed selection not working with space key in `form-single-select`.
```

Author: updated

> * Fixed
>    * Fixed selection not working with space key in `form-single-select`.
>   * Fixed enter key not expanding the combo box in `form-single-select` .

Reviewer 2: ```suggestion

* Fixed
   * Fixed selection not working with space key in `form-single-select`.
   * Fixed enter key not expanding the combo box in `form-single-select` .
```

Author: Updated.

>   * Fixed Enter did not work to expand the combo box example `form-single-select` .

Reviewer: ```suggestion
  * Fixed enter key not expanding the combo box in `form-single-select` .
```

Author: updated
